lib/graphql/middlewares/only-event-owner.ts

```diff
@@ -0,0 +1,36 @@
+
+import { GraphQLError } from "graphql";
+import { TContext } from "../interfaces/context.interface";
+import { MiddlewareFn } from "type-graphql";
+import { getSession } from "next-auth/react";
+
+export const onlyEventOwner: MiddlewareFn<TContext> = async({args, context}, next) => {
+    const session = await getSession(context);
+
+    if ( !session ) { 
+        throw new GraphQLError('Login required', 
+        {
+            extensions: {
+                code: "LOGIN_REQUIRED",
+            },
+        })
+    }
+    const owner_Profile = await context.prisma.profile.findFirst ({
```

Comment:
Probably better to reject request if `owner_Profile` is null. 
nit: rewrite `owner_Profile` to `ownerProfile`